Detectors/TRD/simulation/include/TRDSimulation/Digitizer.h

```diff
@@ -82,6 +82,15 @@ class Digitizer
   double mLastTime = 1.0e10; // starts in the future
   int mEventID = 0;
   int mSrcID = 0;
+  int kNpad = 0;         // Number of pads included in the pad response
+  float kAmWidth = 0;    // Width of the amplification region
+  float kDrWidth = 0;    // Width of the drift retion
+  float kDrMin = 0;      // Drift + Amplification region
+  float kDrMax = 0;      // Drift + Amplification region
+  int timeBinTRFend = 0; // time bin TRF ends
+  int nTimeTotal = 30;   // PLEASE FIX ME when CCDB is ready
```

Comment:
In reference to the comment, I understand this is never going to come from CCDB? 
Its also in TRDCommonParam.h  const expr int kTimeBins = 30;
https://github.com/AliceO2Group/AliceO2/blob/424587b20e58e6867c55be4af37f133e2529dc44/Detectors/TRD/base/include/TRDBase/TRDCommonParam.h#L23

Reply:
The number of time bins that the TRAP process can be set in the TRAPconfig. Therefore it should come out of the CCDB. 

In the end, the simulation would be anchored to an existing run, with a specific TRAP configuration. We should pull the number of time bins out of there.

Note that in the pile-up implementation this would only affect the digits. The signals should extend over whatever time is necessary to cover the tails.

Reply:
This variable is reset when `setSimulationParams()` is called. It actually comes from `o2::trd::kTimeBins` which is defined in `TRDCommonParam.h`, as Sean pointed out.

I agree with Tom's comment, too; I will update the comment in the code to avoid forgetting about this. Probably better, I should add another variable to-be-read from the CCDB, like `nTimeBinsTRAP`, for the signal to ADC processing.

Reply:
The current TrapConfig is out of the Calibrations class.
It is currently back in in the version I am trying to finalise for a pull request as I needed the onlinegaintables and trapconfig coming from ccdb in the TRDTrapSimulatorSpec initialisation routines. The "running" trapconfig is a combination of the CCDB trapconfig and the CCDB onlinegraintables.

There are however 2 things here. We agreed in Cape Town last year that the #timebins would remain 30 and if we change them it will only go down. So although the actual used #timebins will come from what ever is set to run in the trap chips, the data structures will still hold 30.
Did I misunderstand?

Reply:
We agreed to follow the requirement of sending to the DPL containers with elements of fixed size. We have been using 30 time bins as a maximum number of time bins since then, but we knew that we might be running at 22 time bins, for instance.

Reply:
Ok thats as I thought, so having a parameter in TRDCommonParam is right then, and constant. The parameter from the TrapConfig will vary in C13CPUA register, and read in as and when needed.